### PR TITLE
WIP: Fix/improve listeners registration checks

### LINKED ISSUE #19: [Common] Better validation for @Listeners injectable
The validation for these should check if they implement their correct interfaces

These are ofcourse platform specific, so somehow the checks should happen on each platform independently instead of from the common module

### PR DESCRIPTION
**Pull request by mechoriet**
wip needs some work on logging of it

### CODE REVIEW

File: bungee/src/main/java/dev/endoy/helpers/bungee/BungeeEndoyApplication.java [hunk 1]
```diff
@@ -53,6 +55,12 @@ public File getDataFolder()
     @Override
     public void registerListeners( Object listenersInstance )
     {
+        if ( Arrays.stream( listenersInstance.getClass().getDeclaredMethods() )
```
Comment (dieterblancke) [inline bungee/src/main/java/dev/endoy/helpers/bungee/BungeeEndoyApplication.java]: ooohh nice idea to do this here

perhaps it's also a good idea to check if the class implements Listener here?

File: spigot/src/main/java/dev/endoy/helpers/spigot/SpigotEndoyApplication.java [hunk 2]
```diff
@@ -52,6 +54,12 @@ public File getDataFolder()
     @Override
     public void registerListeners( Object listenersInstance )
     {
+        if ( Arrays.stream( listenersInstance.getClass().getDeclaredMethods() )
```
Comment (dieterblancke) [inline spigot/src/main/java/dev/endoy/helpers/spigot/SpigotEndoyApplication.java]: perhaps it's also a good idea to check if the class implements Listener here?

**Comment by dieterblancke** [comment]
maybe only check methods that have public visibility to avoid unnecessary skipping of listeners and warnings

if a Listener class has another method that is used to have less repetitive code, this code will make it not register

File: bungee/src/main/java/dev/endoy/helpers/bungee/BungeeEndoyApplication.java (hunk 1 above)
Comment (mechoriet) [reply bungee/src/main/java/dev/endoy/helpers/bungee/BungeeEndoyApplication.java]: Should be there now

**Comment by mechoriet** [comment]
> maybe only check methods that have public visibility to avoid unnecessary skipping of listeners and warnings
> 
> if a Listener class has another method that is used to have less repetitive code, this code will make it not register

I think in the latest it should only check public methods
maybe later when it becomes a issue we could check the method name if it starts with `on` for example
to narrow it down more

**Comment by dieterblancke** [comment]
> > maybe only check methods that have public visibility to avoid unnecessary skipping of listeners and warnings
> > if a Listener class has another method that is used to have less repetitive code, this code will make it not register
> 
> I think in the latest it should only check public methods maybe later when it becomes a issue we could check the method name if it starts with `on` for example to narrow it down more

Fair enough, if we run into it, we could always change it to still register the class but (debug?) log that certain public methods didn't have the @EventHandler etc annotations

and/or only skip registering the class if there are ZERO eventhandlers/subscriptions in the class

**Review by dieterblancke — APPROVED** [review]

**Comment by dieterblancke** [comment]
LGTM, only possible remark is to maybe add // TODO's on the `System.out.println`'s so when we finally fix the Loggers, we can use those

**Pull request merged**